lib/orderbook/OrderBook.ts

```diff
@@ -286,12 +286,27 @@ class OrderBook extends EventEmitter {
 
     const pairInstance = await this.repository.addPair(pair);
     this.pairInstances.set(pairInstance.id, pairInstance);
-    this.tradingPairs.set(pairInstance.id, new TradingPair(this.logger, pairInstance.id, this.nomatching));
+    this.addTradingPair(pairInstance.id);
 
     this.pool.updatePairs(this.pairIds);
     return pairInstance;
   }
 
+  private addTradingPair = (pairId: string) => {
+    const tp = new TradingPair(this.logger, pairId, this.nomatching);
+    this.tradingPairs.set(pairId, tp);
+    tp.on('peerOrder.dust', (order) => {
```

Comment:
Should we also add code to clean up these listeners before shutting down?

Reply:
I don't believe it's necessary as far as graceful xud shutdowns go, and the `OrderBook` class currently doesn't have a close/shutdown method. `OrderBook` already listens to `Pool` and `Swaps` events that don't get explicitly removed and doesn't have any issues, so I was going to leave that alone. I know in other spots we are explicitly removing listeners (even when not necessary) but I stuck to the pattern in this class. Let me know if that sounds good to you.